ports/soh/Ship of Harkinian.sh

```diff
@@ -20,14 +20,14 @@ source $controlfolder/device_info.txt
 [ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
 
 # Set variables
-GAMEDIR="$directory/ports/soh"
+GAMEDIR="/$directory/ports/soh"
 > "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
 
 cd $GAMEDIR
 
 # Exports
 export LD_LIBRARY_PATH="$GAMEDIR/libs:/usr/lib":$LD_LIBRARY_PATH
-export SDL_GAMECONTROLLERCONFIG=$sdl_controllerconfig
+export SDL_GAMECONTROLLERCONFIG_FILE=$sdl_controllerconfig
```

Comment:
This is not a file, it is an actual controller config definition in the variable `$sdl_controllerconfig`, we already set `SDL_GAMECONTROLLERCONFIG_FILE` already in `control.txt`. 😄 

```suggestion
export SDL_GAMECONTROLLERCONFIG=$sdl_controllerconfig
```

Reply:
So to verify this, I output the env vars in the Ship of Harkinian.sh:
echo "Controller config: $sdl_controllerconfig"
echo "SDL_GAMECONTROLLERCONFIG before: $SDL_GAMECONTROLLERCONFIG"
export SDL_GAMECONTROLLERCONFIG=$sdl_controllerconfig

And this is the output on RG40XX running Knulli:
Controller config: 
SDL_GAMECONTROLLERCONFIG before: 19000000010000000100000000010000,ANBERNIC-keys,platform:Linux,b:b3,a:b4,dpdown:h0.4,leftx:a0,lefty:a1,rightx:a2,righty:a3,lefttrigger:b13,leftstick:b12,dpleft:h0.8,rightshoulder:b8,leftshoulder:b7,righttrigger:b14,rightstick:b15,dpright:h0.2,back:b9,start:b10,dpup:h0.1,y:b6,x:b5,guide:b11,

If you look at this: https://github.com/PortsMaster/PortMaster-GUI/blob/main/PortMaster/knulli/control.txt
And compare with this: https://github.com/PortsMaster/PortMaster-GUI/blob/main/PortMaster/muos/control.txt

It seems like $sdl_controllerconfig is only set on muOS, and not on Knulli. The other question is why are there two variables $sdl_controllerconfig and $SDL_GAMECONTROLLERCONFIG, and why is the latter set both in control.txt *and* the port .sh
To me this looks confusing and prone to error.

Reply:
A compromise for now may be to only use $sdl_controllerconfig if it is set to something, see my second commit below.

I still feel there could be a cleaner solution though, like maybe not setting SDL_GAMECONTROLLERCONFIG in control.txt, if it is only used for a small number of ports

Reply:
There was a recent fix for 40xx on knulli i think? I have yet to do a release as i have been fixing other stuff. So the gamecontroller settings will not be picked up for your device.

Reply:
> There was a recent fix for 40xx on knulli i think?

Where?

Reply:
> > There was a recent fix for 40xx on knulli i think?
> 
> Where?

Recent... 3 weeks ago. lol

https://github.com/PortsMaster/PortMaster-GUI/commit/6aba50b533c50bcc2642c7686bb36ea171c75aaa

Reply:
So Knulli sets a correct SDL_GAMECONTROLLERCONFIG entry. Then we overwrite it with an empty string in the port .sh file. Is this what's happening here?